src/Avalonia.Diagnostics/Diagnostics/DevTools.cs

```diff
@@ -108,14 +103,29 @@ internal static IDisposable Attach(Application? application, DevToolsOptions opt
             return result;
         }
 
-        private static IDisposable Open(Application? application, DevToolsOptions options, Window? owner = default)
+        private static IDisposable Open(Application? application, DevToolsOptions options, Visual? visual = default)
         {
+            var visualOwner = TopLevel.GetTopLevel(visual);
+
             var focussedControl = KeyboardDevice.Instance?.FocusedElement as Control;
             if (application is null)
             {
                 throw new ArgumentNullException(nameof(application));
             }
-            if (s_open.TryGetValue(application, out var window))
+            AvaloniaObject? root;
+            AvaloniaObject parent;
+            if (application.ApplicationLifetime is null)
+            {
+                root = visualOwner;
+                parent = visualOwner ?? TopLevel.GetVisualRoot()!;
```

Comment:
I think we can just throw in this situation, instead of adding hacks for the single visual root.
I.e., throw in situation when users sets up devtools on the whole application without having a lifetime.

Reply:
Lifetime-less application doesn't impose any limits on the number of toplevels